src/modules/keyserver/key-server.repository.ts

```diff
@@ -64,4 +64,16 @@ export class SequelizeKeyServerRepository implements KeyServerRepository {
       defaults: data,
     });
   }
+
+  async update(
+    userId: UserAttributes['id'],
+    data: Partial<KeyServerAttributes>,
+  ) {
+    const [keyServer] = await this.findUserKeysOrCreate(userId, data);
+    if (!keyServer) {
```

Comment:
I would let the use case handle this, as it is the use case imposing that you need to have the keys before updating. In some cases, this could not be important and therefore, not managed by the use case, but in this case, it is vital to ensure and report to the user that the keys are missing.